src/formatDate.js

```diff
@@ -50,7 +50,43 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const objDate = {};
+  const currentDate = date.split(`${fromSeparator}`);
+  const toSeparator = toFormat[3];
+
+  fromFormat.forEach((format, index) => {
+    if (format !== fromSeparator) {
+      objDate[format] = currentDate[index];
+    }
+  });
```

Comment:
Building the objDate directly from the fromFormat and currentDate might lead to potential issues if the toFormat requires a different year format (YYYY vs YY). Consider handling the year conversion separately to ensure correct mapping.